sky/packages/sky/lib/src/rendering/paragraph.dart

```diff
@@ -59,22 +59,27 @@ class RenderParagraph extends RenderBox {
     textPainter.minHeight = constraints.minHeight;
     textPainter.maxHeight = constraints.maxHeight;
     textPainter.layout();
+    // By default, we shrinkwrap to the intrinsic width.
+    double width = constraints.constrainWidth(textPainter.maxIntrinsicWidth);
```

Comment:
> Oh my. So we're doing two layouts every time we want to get any metrics out of text? This seems unnecessarily expensive if all we're trying to do is position the text.

The TextPainter will filter out redundant layouts, which is the common case.

> I don't understand the underlying problem.

The underlying problem is that we want `Text` to layout to its intrinsic width.  That means we need to first compute the intrinsic width and the layout the text to that width.  It's similar to what the C++ render tree would do for absolutely positioned text.

We can change the interface between Dart and C++ to make this work differently if we like.  For example, instead of using a "width: auto" container for the inlines, we can use a C++ container that sizes using intrinsic width.  The net result will be roughly the same, but the logic will just be buried deeper in the system.